bundle/edu.gemini.seqexec.odb/src/main/scala/edu/gemini/seqexec/odb/SeqExecService.scala

```diff
@@ -22,13 +32,77 @@ trait SeqExecService {
 }
 
 object SeqExecService {
-  def client(peer: Peer): SeqExecService = new SeqExecService {
-    val trpc = TrpcClient(peer).withoutKeys
+  val ConnectTimeout: Duration =
+    Duration.ofSeconds(20)
+
+  val ReadTimeout: Duration =
+    Duration.ofSeconds(0)  // 0 is "infinite"
+
+  private def url(peer: Peer, oid: SPObservationID): TrySeq[URL] =
+    catchingAll {
+      new URL(s"http://${peer.host}:8442/ocs3/fetch/pio/${oid.stringValue}")
+    }
+
+  private def open(url: URL): TrySeq[HttpURLConnection] =
+    catchingAll {
+      val conn = url.openConnection.asInstanceOf[HttpURLConnection]
+      conn.setConnectTimeout(ConnectTimeout.toMillis.toInt)
+      conn.setDoOutput(false)
+      conn.setDoInput(true)
+      conn.setReadTimeout(ReadTimeout.toMillis.toInt)
+      conn
+    }
+
+  private def read(conn: HttpURLConnection, oid: SPObservationID): TrySeq[String] =
+    trySeq {
+      conn.getResponseCode match {
+        case HTTP_OK        =>
+          closing(new BufferedReader(new InputStreamReader(conn.getInputStream))) {
+            _.lines.collect(Collectors.joining("\n")).right
+          }
+
+        case HTTP_NOT_FOUND =>
+          MissingObservation(oid).left
+
+        case x              =>
+          val msg = s"Unexpected response code: $x${Option(conn.getResponseMessage).map(m => s": $m").orZero}"
+          SeqException(new RuntimeException(msg)).left
+      }
+    }
+
+  private def parse(xml: String): TrySeq[ISPObservation] =
+    trySeq {
+      val db = DBLocalDatabase.createTransient
+      try {
+        val is = new SpImportService(db)
+        is.importProgramXml(new StringReader(xml)) match {
+          case scala.util.Success(p) => p.getAllObservations.asScala.head.right
+          case scala.util.Failure(e) => SeqException(e).left
+        }
+      } finally {
+        db.getDBAdmin.shutdown
+      }
+    }
 
-    private def call[A](op: SeqExecService => TrySeq[A]): TrySeq[A] =
-      trpc { remote => op(remote[SeqExecService]) }.valueOr(ex => SeqException(ex).left)
+  private def extractSequence(obs: ISPObservation): TrySeq[ConfigSequence] =
+    catchingAll {
+      ConfigBridge.extractSequence(obs, null, IDENTITY_MAP, true)
+    }
 
-    override def sequence(oid: SPObservationID): TrySeq[ConfigSequence] =
-      call(_.sequence(oid))
-  }
+  /** Constructs a SeqExecService that works with a servlet running in the ODB.
+    * When a sequence is requested, it contacts the servlet and requests the
+    * observation XML (wrapped in a program shell suitable for importing). The
+    * XML is then parsed and the sequence is extracted.
+    */
+  def client(peer: Peer): SeqExecService =
```

Comment:
I guess this will be the main client facing API right?
Could we add one that returns `ISPObservation` so we can read more data like the observation name? Can we get `ConfigSequence` from `ISPObservation`?